### PR TITLE
Fix all the stuff you broke last night...

### PR DESCRIPTION
**Pull request by waveform80**
In particular the `pi_revision` thing in PiGPIOPin, all the stuff @lurch
picked up in `pins/data.py` (thank goodness *someone's* watching!), and
make all those links pointing to "Notes" point somewhere useful like
"Pin Numbering"...

### CODE REVIEW

File: gpiozero/pins/data.py [hunk 1]
```diff
@@ -573,10 +573,12 @@ def _parse_pi_revision(revision):
             '3B': True,
             }.get(model, False)
         csi = {
-            'Zero': 0 if pcb_revision == '0.0' else 1,
+            'Zero': 0 if pcb_revision == '1.0' else 1,
             'CM':   2,
             }.get(model, 1)
-        dsi = csi
+        dsi = {
+            'Zero': csi if pcb_revision == '1.0' else 0,
```
Comment (lurch) [inline gpiozero/pins/data.py]: Huh? *Both* revisions of the Pi Zero have no DSI connectors...

Comment (waveform80) [inline gpiozero/pins/data.py]: Urgh ... not thinking straight - bouncing between two branches now (a cherry picked one and the servo one)! Thanks for the catch :)

**Comment by lurch** [comment]
Should the PWM stuff have been in a separate PR?

File: docs/examples/robot_keyboard_2.py [hunk 2]
```diff
@@ -12,9 +12,9 @@
 must_have = {i for i in range(1, 32)}
 must_not_have = {0}
 devices = [
-    device
-    for device in devices
-    for keys in (set(device.capabilities().get(ecodes.EV_KEY, [])),)
+    dev
+    for dev in devices
+    for keys in (set(dev.capabilities().get(ecodes.EV_KEY, [])),)
```
Comment (lurch) [inline docs/examples/robot_keyboard_2.py]: Heh, I was more referring to the multiple usages of `devices` and nested list comprehensions - if I was writing this myself I might have done something like:
```keyboard_devices = filter(is_keyboard, evdev_devices)```
IYSWIM  \*shrug\*

**Comment by waveform80** [comment]
Ahh, I was considering it, but I'm too squeezed for time right now so this is basically "fix everything so you've got a clean PR which only includes Servo et al" - as I'm anticipating that's going to generate a bit of discussion.

Incidentally, there's a Raspbian freeze coming up fairly shortly (@bennuttall tomorrow?) so in a vague effort to get GPIO Zero (and some other stuff) into the next Raspbian image I'm going to try and get a 1.3 release out rather shortly. As a result, the Servo stuff is going to be pared down to the bare minimum that I think is *definitely* fine (on the assumption that adding stuff is always easier than removing stuff as the latter is backwards incompatible) - I'm also contemplating merging ButtonBoard this evening if you're happy for that to go in as it is?

**Comment by lurch** [comment]
Yeah, feel free to merge in whatever you think's suitable - afraid I don't have much time for working on gpiozero ATM :-/

File: docs/examples/robot_keyboard_2.py (hunk 2 above)
Comment (waveform80) [inline docs/examples/robot_keyboard_2.py]: Ah, that would be neater ... maybe next time after a few users have posted "WTF?!" tickets next to this recipe :)

**Comment by waveform80** [comment]
I know the feeling - still, very much appreciate you taking the time to pick up on all my bugs :)

**Pull request merged**